inc/features/blocks/namespace.php

```diff
@@ -69,3 +78,248 @@ function get_block_settings( string $name ) : ?array {
 
 	return $settings;
 }
+
+/**
+ * Register REST API endpoints for Experience Blocks.
+ *
+ * @return void
+ */
+function rest_api_init() : void {
+
+	// Experience blocks views endpoint.
+	register_rest_route( 'analytics/v1', 'xbs/(?P<id>[a-z0-9-]+)/views', [
+		[
+			'methods' => WP_REST_Server::READABLE,
+			'callback' => __NAMESPACE__ . '\\handle_views_request',
+			'permission_callback' => __NAMESPACE__ . '\\check_views_permission',
+			'args' => [
+				'id' => [
+					'description' => __( 'The experience block client ID', 'altis-experiments' ),
+					'required' => true,
+					'type' => 'string',
+					'validate_callback' => __NAMESPACE__ . '\\validate_id',
+					'sanitize_callback' => __NAMESPACE__ . '\\sanitize_id',
+				],
+				'post_id' => [
+					'description' => __( 'An optional post ID to filter by.', 'altis-experiments' ),
+					'type' => 'number',
+				],
+			],
+		],
+		'schema' => [
+			'type' => 'object',
+			'properties' => [
+				'total' => [ 'type' => 'number' ],
+				'audiences' => [
+					'type' => 'array',
+					'items' => [
+						'type' => 'object',
+						'properties' => [
+							'id' => [ 'type' => 'number' ],
+							'count' => [ 'type' => 'number' ],
+						],
+					],
+				],
+				'posts' => [
+					'type' => 'array',
+					'items' => [
+						'type' => 'object',
+						'properties' => [
+							'id' => [ 'type' => 'number' ],
+							'count' => [ 'type' => 'number' ],
+							'audiences' => [
+								'type' => 'array',
+								'items' => [
+									'type' => 'object',
+									'properties' => [
+										'id' => [ 'type' => 'number' ],
+										'count' => [ 'type' => 'number' ],
+									],
+								],
+							],
+						],
+					],
+				],
+				'postId' => [ 'type' => 'number' ],
+			],
+		],
+	] );
+}
+
+/**
+ * Validate Experience Block ID.
+ *
+ * @param string $param The Experience Block ID.
+ * @return bool
+ */
+function validate_id( $param ) : bool {
+	return (bool) preg_match( '/[a-z0-9-]+/', $param );
+}
```

Comment:
No function for this already I guess? Wack.

Reply:
`sanitize_uuid()`? nah doesn't seem so